File: server/TodoApp/TodoApp.Domain/Models/Task.cs
namespace TodoApp.Domain.Models;

public class Task : BaseEntity
{
   public string Title { get; set; }
   public string? Note { get; set; }
   public bool IsCompleted { get; set; }
   public DateTime? DueTime { get; set; }
   
   public Guid UserId { get; set; }
   public User User { get; set; }
   public List<TaskCategory> TaskCategories { get; set; }
}

File: web/to-do-app-web/src/app/core/types/task/task.ts
import { Category } from '../category/category';

export interface Task {
    id: string;
    title: string;
    note: string | null;
    createdAt: Date | string;
    isCompleted: boolean;
    dueTime: Date | string;
    taskCategories: Category[];
}

export interface SearchTaskResponse {
    items: Task[];
    pageSize: number;
    page: number;
    totalCount: number;
    totalPages: number;
}